core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -211,6 +376,11 @@ public HttpResponse doCompleteInstall() throws IOException, ServletException {
     static void completeSetup(Jenkins jenkins) {
         jenkins.setInstallState(InstallState.INITIAL_SETUP_COMPLETED);
         InstallUtil.saveLastExecVersion();
+        try {
+            completeUpgrade(jenkins);
+        } catch(IOException ex) {
+            LOGGER.log(Level.SEVERE, "Cannot update the upgrade status", ex);
```

Comment:
would make sense to add an administrative monitor or something similar. Subject for the future improvement IMHO